lib/eslint.js

```diff
@@ -505,6 +506,46 @@ function stripUnicodeBOM(text) {
     return text;
 }
 
+/**
+ * Wraps a rule action with an exception handler.
+ *
+ * The exception handler will catch exceptions thrown by rule actions, decorate
+ * the exception with rule metadata, and rethrow the exception. This will allow
+ * us to detect the uncaught exception as a rule crash and show a user-friendly
+ * error message.
+ *
+ * @param {function} action The rule action to wrap.
+ * @param {string} ruleName The name of the rule.
+ * @returns {function} A function which will catch, decorate, and rethrow rule exceptions.
+ */
+function createRuleExceptionHandler(action, ruleName) {
+    var pluginDelimiterIndex = ruleName.indexOf("/"),
+        pluginName = null;
+
+    if (pluginDelimiterIndex > -1) {
+        pluginName = ruleName.slice(0, ruleName.indexOf("/"));
+    }
+
+    return function() {
+        try {
+            action.apply(null, arguments);
+        } catch (err) {
+            err.originalMessage = err.message;
+            err.message = "Rule \"" + ruleName + "\" threw an exception.";
+
+            err.messageTemplate = "rule-crashed";
+            err.messageData = {
+                ruleName: ruleName,
+                pluginName: pluginName,
+                eslintVersion: ESLINT_VERSION,
+                stackTrace: err.stack
```

Comment:
I'm already seeing that I never actually use the originalMessage bit. I probably should.